runners/spark/src/main/java/org/apache/beam/runners/spark/translation/TransformTranslator.java

```diff
@@ -146,32 +146,36 @@ public void evaluate(GroupByKey<K, V> transform, EvaluationContext context) {
 
         JavaRDD<WindowedValue<KV<K, Iterable<V>>>> groupedByKey;
         Partitioner partitioner = getPartitioner(context);
-        // As this is batch, we can ignore triggering and allowed lateness parameters.
-        if (windowingStrategy.getWindowFn().equals(new GlobalWindows())
-            && windowingStrategy.getTimestampCombiner().equals(TimestampCombiner.END_OF_WINDOW)) {
-          // we can drop the windows and recover them later
-          groupedByKey =
-              GroupNonMergingWindowsFunctions.groupByKeyInGlobalWindow(
-                  inRDD, keyCoder, coder.getValueCoder(), partitioner);
```

Comment:
Actually, seems that these optimizations are not mutually exclusive. We might want to apply both.